spec/models/message_spec.rb

```diff
@@ -13,6 +13,25 @@
     end
   end
 
+  describe "criteria_rules validation" do
+    it "is valid with criteria_rules are valid" do
+      message = build(:message, criteria_rules: [
```

Comment:
Feels like a slight overlap of concern for the tests below compared with tests in CriteriaSchemaValidator but I suppose that's okay as this is explicitly testing the validity of the record instead of the attribute.

Reply:
Yup it is, we should test that criteria rules are validated so either we test with an example thing that overlaps or use a test that checks whether the validator is called. I prefer the former option since it's a real test but this is an area where there are differing opinions.

Reply:
Nice, good to know! 👍